src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -433,13 +434,30 @@ public function getRelationValue($key)
             return $this->relations[$key];
         }
 
+        if (! $this->isRelation($key)) {
+            return;
+        }
+
+        if ($this->withStrictLoading) {
+            throw new StrictLoadingViolationException($this, $key);
+        }
+
         // If the "attribute" exists as a method on the model, we will just assume
         // it is a relationship and will load and return results from the query
         // and hydrate the relationship's value on the "relationships" array.
-        if (method_exists($this, $key) ||
-            (static::$relationResolvers[get_class($this)][$key] ?? null)) {
-            return $this->getRelationshipFromMethod($key);
-        }
+        return $this->getRelationshipFromMethod($key);
+    }
+
+    /**
+     * Determine if the given key is a relation.
+     *
+     * @param  string  $key
+     * @return bool
+     */
+    protected function isRelation($key)
```

Comment:
I am thinking of changing this to public method, because imo, I might use this method elsewhere in my code in the future.